freetextresponse/private/view.js

```diff
@@ -4,46 +4,109 @@ function FreeTextResponseView(runtime, element) {
     var $ = window.jQuery;
     var $element = $(element);
     var buttonSubmit = $element.find('.check.Submit');
-    var textareaParent = $element.find('.student_answer').parent();
     var usedAttemptsFeedback = $element.find('.action .used-attempts-feedback');
     var problemProgress = $element.find('.problem-progress');
-    var submitParent = $element.find('.Submit').parent();
-    var wordCountError = $element.find('.word-count-error');
     var submissionReceivedMessage = $element.find('.submission-received');
-    var url = runtime.handlerUrl(element, 'submit');
     var userInputClass = 'user-input';
+    var buttonSave = $element.find('.save');
+    var userAlertMessage = $element.find('.user_alert');
+    var textareaStudentAnswer = $element.find('.student_answer');
+    var textareaParent = textareaStudentAnswer.parent();
+    var url = runtime.handlerUrl(element, 'submit');
+    var urlSave = runtime.handlerUrl(element, 'save_reponse');
+   
+
+    // Override notify if it does not exist. Like in the xblock workbench.
+    // The "or" (`||`) makes sure we don't redefine the method if it already exists.
+    runtime.notify = runtime.notify || function () {
+        // The logging statement is optional.
```

Comment:
Again, these were overly annotated examples for the purposes of explaining and justifying my review.